### PR TITLE
Add cl_intel_unified_shared_memory to supported extensions

### PR DESCRIPTION
**Pull request by pcolberg**
The SYCL runtime uses but currently does not check for this extension.

From a cursory glance all new API functions are implemented.

https://www.khronos.org/registry/OpenCL/extensions/intel/cl_intel_unified_shared_memory.html

### CODE REVIEW

**Review by sherry-yuan — APPROVED** [review]
LGTM

**Review by zibaiwan — APPROVED** [review]
Thanks Peter!

**Comment by pcolberg** [comment]
Rerunning the Windows build workflows after #96 was merged does not update the base branch commit into which the pull-request branch is merged, which means that the workflows still fail. I have not pushed a rebased PR since that would dismiss reviewer approval. Instead I will be trying the repository setting [Always suggest updating pull request branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-suggestions-to-update-pull-request-branches) to see whether it retains reviewer approval.

**Comment by pcolberg** [comment]
Enable the repository setting [Always suggest updating pull request branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-suggestions-to-update-pull-request-branches) enabled the Update branch button, which merges the up-to-date main branch into the pull request branch. This is not quite the same as manual rebasing, but it retains reviewer approval. Merge with rebase then removes the merge commit again.

**Pull request merged**
